app/models/user.rb

```diff
@@ -550,6 +550,26 @@ def close_account!
     AccountDeletion.create(:person => self.person)
   end
 
+  def lock_access!(opts={})
+    if opts[:unlock_in]
+      self.lock_expires = true
+      self.locked_at = Time.now.utc + opts[:unlock_in]
+      save
+    else
+      self.lock_expires = false
+      super
+    end
+  end
+
+  def lock_expired?
```

Comment:
> Have you checked that this actually gets called?

yes it does. Its being called while authenticating.. I had no desire to read through the devise source code